### PR TITLE
rename dropdown description to label for consistency with other input components

### PR DESCRIPTION
**Pull request by hwigmore**
breaking change, renaming a prop from `description` to `label` for consistency with textfield

fyi i'm leaving for a week so feel free to add to this and/or merge

@emma-floored

### CODE REVIEW

File: packages/blocks-react/dropdowns/Dropdown.jsx [hunk 1]
```diff
@@ -27,12 +27,8 @@ const DropdownMenu = (props) => {
     <div className="blx-dropdown-wrapper" ref={props.forwardedRef}>
       <div className="blx-dropdown">
 
-        {/* DESCRIPTION */}
-        { props.description &&
-            <div>
```
Comment (hwigmore) [inline packages/blocks-react/dropdowns/Dropdown.jsx]: by removing this surrounding div, the label margin-bottom actually gets applied and therefore the space between label and input is consistent with textfield

**Comment by hwigmore** [comment]
@hamham91 also tagging you on this, we'll need this pulled in to I&L to get the proper spacing between dropdown label and input

File: packages/blocks-react/dropdowns/Dropdown.jsx [hunk 2]
```diff
@@ -111,9 +107,9 @@ DropdownMenu.defaultProps = {
   options: [],
   text: 'Choose an option',
   icon: null,
-  description: '',
+  label: '',
   value: null,
-  onChange: () => {},
+  onSelect: () => {},
```
Comment (emma-floored) [inline packages/blocks-react/dropdowns/Dropdown.jsx]: does this change fix that other bug you mentioned?

Comment (hwigmore) [reply packages/blocks-react/dropdowns/Dropdown.jsx]: so i'm not 100% sure that i was right about this bug. i need to revisit this. 

this change was just changing to what we actually expect in this file (if you search the file for `onChange` you'll see it's only mentioned in propTypes, but `onSelect` is actually the prop that's used in render

related: i would love to get linter working for blocks!!

Comment (hwigmore) [reply packages/blocks-react/dropdowns/Dropdown.jsx]: but i think my original bug / confusion stems from `keyControlledMenu` where there's a component function called `onSelect` that either calls `onClick` or `onChange`. i'll take a look again later, i might just be confused!

**Comment by emma-floored** [comment]
👍

**Pull request merged**